restorer.go

```diff
@@ -35,13 +39,25 @@ func (r *Restorer) Restore(cache Cache) error {
 
 	var g errgroup.Group
 	for _, buildpack := range r.Buildpacks {
+		cachedLayers := meta.MetadataForBuildpack(buildpack.ID).Layers
+
+		var cachedFn func(bpLayer) bool
+		if api.MustParse(buildpack.API).Compare(api.MustParse("0.6")) >= 0 {
+			cachedFn = func(l bpLayer) bool {
+				layer, ok := cachedLayers[launch.EscapeID(filepath.Base(l.path))]
+				return ok && layer.Cache
+			}
+		} else {
+			cachedFn = forCached
+		}
+
 		buildpackDir, err := readBuildpackLayersDir(r.LayersDir, buildpack, r.Logger)
 		if err != nil {
 			return errors.Wrapf(err, "reading buildpack layer directory")
 		}
+		foundLayers := buildpackDir.findLayers(cachedFn)
```

Comment:
This could be inlined but it was useful for debugging

Reply:
I agree. I like having it as a separate line.